python/tk_multi_publish2/api/item.py

```diff
@@ -0,0 +1,1012 @@
+        """
+        return self._description
+
+    @description.setter
+    def description(self, new_description):
+        """Sets a new description for the item with the given string."""
+        self._description = new_description
+
+    @property
+    def enabled(self):
+        """
+        Boolean property which indicates whether this item and its children
+        should be enabled within a publish UI.
+        """
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, enabled):
+        # setter for enabled
+        self._enabled = enabled
+
+    @property
+    def expanded(self):
+        """
+        Boolean property which indicates whether this item should be expanded to
+        show its children when shown in a publish UI.
+        """
+        return self._expanded
+
+    @expanded.setter
+    def expanded(self, is_expanded):
+        """Setter for the expanded property."""
+        self._expanded = is_expanded
+
+    @property
+    def icon(self):
+        """
+        The associated icon, as a QPixmap.
+
+        The icon is a small square image used to represent the item visually
+
+        .. image:: ./resources/item_icon.png
+
+        |
+
+        If no icon has been defined for this node, the parent icon is returned,
+        or a default one if not defined
+
+        .. warning:: This property will return ``None`` when run without a UI
+            present
+
+        .. note:: The :py:attr:`~icon` is for use only in the publisher UI and
+            is a small representation of item being published. The icon should
+            not be confused with the item's :py:attr:`~thumbnail` which is
+            typically associated with the resulting published item in Shotgun.
+        """
+        return self._get_image(
+            get_img_path=lambda: self._icon_path,
+            get_pixmap=lambda: self._icon_pixmap,
+            set_pixmap=lambda pixmap: setattr(self, "_icon_pixmap", pixmap),
+            get_parent_pixmap=lambda: self.parent.icon,
+            default_pixmap_path=":/tk_multi_publish2/item.png"
+        )
+
+    def _get_image(self, get_img_path, get_pixmap, set_pixmap, get_parent_pixmap, default_pixmap_path):
+        """
+        Retrieves the image for the icon or thumbnail of this item.
+
+        This method is written in a generic fashion in order to avoid complex logic duplicated
+        inside the class.
+
+        It takes a series of getter and setter methods that allow to update the
+        thumbnail or the icon of this item.
+
+        :param function get_img_path: Function returning the path to an image on disk.
+        :param function get_pixmap: Function returning the pixmap for the image.
+        :param function set_pixmap: Function allowing to set the in-memory pixmap for the image.
+        :param function get_parent_pixmap: Function allowing to get the pixmap of the parent item.
+        :param str default_pixmap_path: Path to the default pixmap.
```

Comment:
[nitpick] A pixmap doesn't have a path, so should this be "default_image_path" instead?